### ISSUE TITLE
Segmentation fault (core dumped) when compiling nested @media in sub-properties

### ISSUE THREAD

**Issue by frantic1048** [sass/node-sass#1667]
I know this is not legal syntax for SCSS, but I think throwing a sensible error is better than one line segfault...

In this case, ruby-sass reports :

`Illegal nesting: Only properties may be nested beneath properties.`

I found an old similar segfault in #1373 , while it is not exactly same as this case.

**node-sass version**: 3.8.0
**libsass version**: 3.3.6
**node version**: 6.3.1
**platform**: Arch Linux latest with linux-zen 4.6.5

Sample code(`style.scss`):

``` scss
body {
  margin: {
    @media (min-width: 30em) { top: 1em; }
  }
}
```

command: 

``` bash
$node_modules/.bin/node-sass src/style.scss dist/style.css
Segmentation fault (core dumped)
```

**Comment by nschonni** [comment]
Can you file an issue with https://github.com/sass/libsass/issues and link it back here, thanks!

**Comment by frantic1048** [comment]
All right, I've filed it on sass/libsass#2148 . Did I link correctly ?

**Comment by nschonni** [comment]
Thanks! Since it's marked as fixed in master it will be addressed with the next libsass bump

**Comment by nschonni** [comment]
@frantic1048 is this fixed for you in 4.0 now?

**Comment by frantic1048** [comment]
@nschonni Works well 👍
